src/app/(post-login)/submission/program/[shortName]/clinical-data/ClinicalEntityDataTable.tsx

```diff
@@ -29,6 +60,132 @@ export type DonorEntry = {
 	[k: string]: string | number | boolean;
 };
 
+const errorColumns = [
+	{
+		accessorKey: 'entries',
+		Header: '# Affected Records',
+		id: 'entries',
+		maxWidth: 135,
+	},
+	{
+		accessorKey: 'fieldName',
+		Header: `Field with Error`,
+		id: 'fieldName',
+		maxWidth: 215,
+	},
+	{
+		accessorKey: 'errorMessage',
+		Header: `Error Description`,
+		id: 'errorMessage',
+	},
+];
+
+const NoDataCell = () => (
+	<div
+		css={css`
+			display: flex;
+			flex-direction: column;
+			align-items: center;
+			justify-content: center;
+			padding: 80px 0;
+		`}
+	>
+		<ContentPlaceholder title="No Data Found.">
+			<img alt="No Data" src={noDataSvg} />
+		</ContentPlaceholder>
+	</div>
+);
+
+const completionKeys = Object.values(aliasSortNames);
+const completionColumnNames = Object.keys(aliasSortNames);
+const emptyCompletion = {
+	DO: 0,
+	PD: 0,
+	FO: 0,
+	NS: 0,
+	TR: 0,
+	TS: 0,
+};
+
+const noDataCompletionStats = [
+	{
+		donor_id: 0,
+		...emptyCompletion,
+	},
+];
+
+const completionColumnHeaders = {
+	donor: 'DO',
+	primaryDiagnosis: 'PD',
+	normalSpecimens: 'NS',
+	tumourSpecimens: 'TS',
+	treatments: 'TR',
+	followUps: 'FO',
+};
```

Comment:
Inclined to suggest abstracting constants into a separate file, and helpers into another one, leaving only the component logic here.